NExtends/Collections/ObjectPropertiesList.cs

```diff
@@ -0,0 +1,362 @@
+		/// <param name="length">Number of properties with specified prefix</param>
+		/// <param name="offset">Couting offset, (e.g. 1 if properties are named field1, field, etc.</param>
+		public ObjectPropertiesListNumberedPropertiesAttribute(int order, string fieldPrefix, int length, int offset)
+		{
+			List<string> propertiesNames = new List<string>();
+
+			for (int i = offset; i < length + offset; i++)
+				propertiesNames.Add(string.Format("{0}{1}", fieldPrefix, i));
+
+			this.IndexableProperties = propertiesNames;
+			this.Order = order;
+		}
+	}
+
+	/// <summary>
+	/// Use this attribute to index selected properties.
+	/// </summary>
+	[AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+	public class ObjectPropertiesListPropertiesAttribute : Attribute, IObjectPropertiesListParams
+	{
+		public IEnumerable<string> IndexableProperties { get; protected set; }
+		public int Order { get; protected set; }
+
+		/// <summary>
+		/// Attribute to index selected properties 
+		/// </summary>
+		/// <param name="order">In case you use multiple attributes, design the order of the properties in the resulting list</param>
+		/// <param name="properties">Name of properties to index</param>
+		public ObjectPropertiesListPropertiesAttribute(int order, params string[] properties)
+		{
+			this.IndexableProperties = properties;
+			this.Order = order;
+		}
+	}
+
+	/// <summary>
+	/// Use expression trees to get property getter and setters
+	/// http://weblogs.asp.net/marianor/archive/2009/04/10/using-expression-trees-to-get-property-getter-and-setters.aspx
+	/// 
+	/// There are times when you need get and set property values and you do not know the type of the properties.
+	/// One option is use reflection through GetValue and SetValue from PropertyInfo class, but this wears to poor performance in our code.
+	/// In order to do that we can use an Expression Tree to generate delegates that allow to get and set the value of the required property, 
+	/// for example building a couple of extensions methods applying to PropertyInfo:
+	/// 
+	/// Modification done to the code online: The getter and setter delegates are strictly typed, by simply adding a second generic parameter T and replacing "object" by T.
+	/// </summary>
+	public static class PropertyInfoExtensions
+	{
+		public static Func<K, T> GetValueGetter<K, T>(this PropertyInfo propertyInfo)
+		{
+			if (typeof(K) != propertyInfo.DeclaringType)
+			{
+				throw new ArgumentException();
+			}
+
+			var instance = Expression.Parameter(propertyInfo.DeclaringType, "i");
+			var property = Expression.Property(instance, propertyInfo);
+			var convert = Expression.TypeAs(property, typeof(T));
+			return (Func<K, T>)Expression.Lambda(convert, instance).Compile();
+		}
+
+		public static Action<K, T> GetValueSetter<K, T>(this PropertyInfo propertyInfo)
+		{
+			if (typeof(K) != propertyInfo.DeclaringType)
+			{
+				throw new ArgumentException();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use a constructor overloads that allows a more meaningful exception message to be provided. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)